test/lit.cfg.py

```diff
@@ -24,6 +24,9 @@
 # suffixes: A list of file extensions to treat as test files.
 config.suffixes = ['.td', '.mlir', '.ll', '.fir', '.sv']
 
+if config.bindings_python_enabled:
+  config.suffixes.append('.py')
```

Comment:
This works to exclude the python tests when the bindings are disabled, but I think it might be better to actually use `config.available_features` and `REQUIRES` in the tests. That seems more idiomatic, but it's mildly annoying to put `REQUIRES` in all the tests.

Reply:
I would really prefer the REQUIRES/features support even upstream, but I wasn't lit-erate enough to know that is what I wanted when starting the project. With directory exclusions (upstream) or extension exclusions (here), it gets a little bit hard to manage once things grow outside of their first location.

If you work this out locally, I can adapt it upstream.

Reply:
> lit-erate

:grimacing: 

I will play around with this and see how it looks here.

Reply:
@stellaraccident I believe this was all it took to make REQUIRES work for Python bindings tests: https://github.com/llvm/circt/pull/727/commits/444570a5699fc910b163266ab04b8ee1b500038d.